lib/cqm/converter/hds_record.rb

```diff
@@ -125,5 +125,17 @@ def to_qdm(record)
 
       patient
     end
+
+    def generate_qdm_data_element(dc_fixed_keys, dc_type)
+      data_element = QDM.const_get(dc_type).new.from_json(dc_fixed_keys.to_json)
+
+      if data_element.is_a?(QDM::EncounterPerformed)
+        data_element.facilityLocations = data_element.facilityLocations.map do |facility|
+          QDM::FacilityLocation.new.from_json(facility.to_json)
+        end
+      end
```

Comment:
I think there might be a few "nested" QDM things we might eventually need to handle here... For now, do you think adding a little comment (something along the lines of: "Any nested QDM types that need initialization should be handled here when converting from the QDM models to the HDS models") makes sense?

Reply:
Sure thing.